modules/web/src/core/workspace/model/folder.js

```diff
@@ -0,0 +1,120 @@
+/**
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+import _ from 'lodash';
+import { getPathSeperator } from 'api-client/api-client';
+import EventChannel from 'event_channel';
+import { EVENTS } from './../constants';
+
+/**
+ * Represents a Folder opened in workspace
+ */
+class Folder extends EventChannel {
```

Comment:
Do we need an event channel for the folder, will it make more sense to handle all the event via app event stream.